examples/hitl/rearrange_v2/rearrange_v2.py

```diff
@@ -388,24 +406,36 @@ def sim_update(self, dt, post_sim_update_dict):
 
         self._check_change_episode()
 
-        reachable_ao_handle = self._find_reachable_ao(
-            self._app_service.gui_agent_controller
-        )
-        if reachable_ao_handle is not None:
-            self._highlight_ao(reachable_ao_handle)
-            if self._app_service.gui_input.get_key_down(GuiInput.KeyNS.Z):
-                self._open_close_ao(reachable_ao_handle)
+        for user_index in range(self._num_users):
+            reachable_ao_handle = self._find_reachable_ao(
+                self._get_gui_agent_translation(user_index)
+            )
+            if reachable_ao_handle is not None:
+                self._highlight_ao(reachable_ao_handle)
+                if self._get_user_key_down(user_index, GuiInput.KeyNS.Z):
+                    self._open_close_ao(reachable_ao_handle)
 
         if not self._paused:
-            self._update_grasping_and_set_act_hints()
+            for user_index in range(self._num_users):
+                self._update_grasping_and_set_act_hints(user_index)
             self._app_service.compute_action_and_step_env()
         else:
             # temp hack: manually add a keyframe while paused
             self.get_sim().gfx_replay_manager.save_keyframe()
 
+        # todo: visualize objects properly for each user (this requires a separate debug_line_render per user!), or find a reasonable debug line visualization that can be shared between both users every frame.
         if self._held_obj_id is None:
             self._pick_helper.viz_objects()
 
+        if (
```

Comment:
For local multi-user testing, use the `T` key to toggle the camera between users. Toggling the camera while holding an object is super-confusing so I disabled this.